### PR TITLE
Week three slides fixes

### PR DESCRIPTION
**Pull request by JasonGibson274**
Still missing speaker notes but lets me know what you think of content

### CODE REVIEW

File: slides/week3/week3.org [hunk 1]
```diff
@@ -2,66 +2,158 @@
...
 - Branching and merging in Git
+- Using light sensor to count grey squares
+
+* What is an object?
+- a module of reusable code
+  - maintains some sort of state
+    - variables
+  - have behaviors
+    - methods
+- can have multiple instances of the same object
```
Comment (jgkamat) [inline slides/week3/week3.org:16]: Did you mean class here? I guess that it's correct both ways though.

File: slides/week3/week3.org [hunk 2]
```diff
@@ -2,66 +2,158 @@
...
-* What's a container?
-- An object you can use to store other objects
-- Objects are identified by keys (which can also be objects)
-- Some important properties:
-  - *Ordered:* the contents are in a specific order
-  - *Associative:* two objects are associated with each other within the container
-  - *Set:* the value of an object is its key
-  - *Unique keys:* no two objects can have the same key
+* Array methods
+| =at=    | returns the value at a specific index |
```
Comment (jgkamat) [inline slides/week3/week3.org:29]: Finally people are aligning tables :tada:

File: slides/week3/week3.org [hunk 3]
```diff
@@ -2,66 +2,158 @@
...
+* Vector methods
 #+BEGIN_NOTES
 Vectors can grow as items are added, whereas arrays are static.
 #+END_NOTES
-- Everything from the last slide, and...
-| =clear= | removes all items |
-| =insert= | inserts an item at a specific index |
-| =push_back= | adds a given element to the end |
-| =pop_back= | removes the last element |
+- Everything from the arrays slide, and...
```
Comment (jgkamat) [inline slides/week3/week3.org]: If you wanted, you could link to the arrays slide here via [org-store-link](http://orgmode.org/manual/Handling-links.html). `C-c l` with point on the header for the arrays slide, and insert the link via `C-c C-l`.

File: slides/week3/week3.org [hunk 4]
```diff
@@ -2,66 +2,158 @@
...
-All of these can be restricted to a range in the container using iterators. All of these are in <algorithm> except for accumulate and iota, which are in <numeric>.
+picture upcoming. mention that the method should be called on the containing object
 #+END_NOTES
-| =sort= | sorts a container |
-| =count= | counts the number of items in a container that match a given item |
-| =find= | returns an iterator to the first element that matches a given item |
+| =*=      | gets the value at the current index                     |
+| =begin=  | starts at the *first* item that moves *up* in indexes   |
+| =end=    | starts at the *last* item that moves *up* in indexes    |
+| =rbegin= | starts at the *first* item that moves *down* in indexes |
```
Comment (jgkamat) [inline slides/week3/week3.org]: This is a bit confusing, at least to me. This makes it look like you need to decrement the index to move it forward. Also, doesen't it [start at the end](http://www.cplusplus.com/reference/vector/vector/rbegin/)?

Maybe:

starts at the last item and moves backwards when incremented.

File: slides/week3/week3.org [hunk 5]
```diff
@@ -2,66 +2,158 @@
...
+| =rend=   | starts at the *last* item that moves *down* in indexes  |
+
+* Moving iterators
+#+BEGIN_NOTES
+picture upcoming
+#+END_NOTES
+- =++= moves the iterator forwards
```
Comment (jgkamat) [inline slides/week3/week3.org]: With reverse iterators, won't this move them backwards?

Perhaps, 'increments the iterator'

File: slides/week3/week3.org [hunk 6]
```diff
@@ -2,66 +2,158 @@
...
+- =--= moves the iterator backwards
+
+* Iterator practice
```
Comment (jgkamat) [inline slides/week3/week3.org]: Perhaps for better export into other formats, you can keep this under the same heading as above.

You can force a split within a heading with `#+REVEAL: split`

[docs](https://github.com/yjwen/org-reveal/#force-split).

**Review by jgkamat — CHANGES_REQUESTED** [review]
Overall looks really good, here are some tiny issues though...

File: slides/week3/week3.org (hunk 1 above)
Comment (JasonGibson274) [reply slides/week3/week3.org:16]: I was trying to avoid saying classes since we have no covered them yet.

File: slides/week3/week3.org [hunk 7]
```diff
@@ -2,66 +2,152 @@
...
 #+END_NOTES
-| =array= | a static-length ordered list of objects |
-| =vector= | a dynamic-length ordered list of objects |
-| =set= | an ordered group of unique objects |
-| =unordered_set= | an unordered group of unique objects |
-| =map= | an ordered group of key-value pairs, with unique keys |
-| =unordered_map= | an unordered group of key-value pairs, with unique keys |
+- an object that can contain a variable number of other objects
+- you can keep adding elements to the vector
+- order of insertion in maintained
```
Comment (chachmu) [inline slides/week3/week3.org]: Found a typo, order of insertion *is* maintained.

**Review by jgkamat — DISMISSED** [review]
Minus @chachmu's note, I think this is good. Let's get this merged soon plz

**Review by jgkamat — APPROVED** [review]
LGTM

**Pull request merged**